Makefile

```diff
@@ -1,104 +1,38 @@
-.PHONY: rel deps test show_test_results
+.PHONY: rel
 LOG=$(subst TARGET,$@,TARGET.log 2>&1 || (cat TARGET.log; exit 1))
 SILENCE_COVER =  | grep -v "logs.*\\.coverdata"
 SILENCE_COVER += | grep -v "Analysis includes data from imported files"
 SILENCE_COVER += | grep -v "WARNING: Deleting data for module"
 LOG_SILENCE_COVER=$(subst TARGET,$@,TARGET.log 2>&1 || (cat TARGET.log $(SILENCE_COVER); exit 1))
 EJABBERD_DIR = apps/ejabberd
-EJD_INCLUDE = $(EJABBERD_DIR)/include
-EJD_PRIV = $(EJABBERD_DIR)/priv
 XEP_TOOL = tools/xep_tool
 EJD_EBIN = $(EJABBERD_DIR)/ebin
-DEVNODES = node1 node2 node3 fed1
-
-all: deps compile
-
-compile: rebar
-	./rebar $(OPTS) compile > $(LOG)
-
-deps: rebar
-	./rebar get-deps > $(LOG)
-
-clean: rebar configure.out
-	rm -rf apps/*/logs
-	. ./configure.out && ./rebar clean
-
-quick_compile: rebar
-	./rebar $(OPTS) compile skip_deps=true > $(LOG)
-
-reload: quick_compile
-	@E=`ls ./rel/mongooseim/lib/ | grep ejabberd-2 | sort -r | head -n 1` ;\
-	rsync -uW ./apps/ejabberd/ebin/*beam ./rel/mongooseim/lib/$$E/ebin/ ;\
-
-reload_dev: quick_compile
-	@for NODE in $(DEVNODES); do \
-		E=`ls ./dev/mongooseim_$$NODE/lib/ | grep ejabberd-2 | sort -r | head -n 1` ;\
-		rsync -uW ./apps/ejabberd/ebin/*beam ./dev/mongooseim_$$NODE/lib/$$E/ebin/ ;\
-	done
-
-# Run a single suite
-# Example to run apps/ejabberd/test/cassandra_SUITE.erl:
-# make ct SUITE=cassandra
-ct: deps quick_compile
-	@(if [ "$(SUITE)" ]; \
-		then ./rebar $(OPTS) ct suite=$(SUITE) skip_deps=true; \
-		else ./rebar $(OPTS) ct skip_deps=true; fi) \
-		> $(LOG_SILENCE_COVER)
-
-# This compiles and runs one test suite. For quick feedback/TDD.
-# Example:
-# $ make qct SUITE=amp_resolver_SUITE
-qct:
-	mkdir -p /tmp/ct_log
-	@if [ "$(SUITE)" ]; then ct_run -pa apps/*/ebin -pa deps/*/ebin -pa ebin -dir apps/*/test\
-        -I apps/*/include -I apps/*/src -logdir /tmp/ct_log -suite $(SUITE)_SUITE -noshell;\
-	else ct_run -pa apps/*/ebin -pa deps/*/ebin -pa ebin -dir apps/*/test\
-        -I apps/*/include -I apps/*/src -logdir /tmp/ct_log -noshell; fi
-
-test: test_deps
-	cd test/ejabberd_tests; make test
-
-test_preset: test_deps
-	cd test/ejabberd_tests; make test_preset
+DEVNODES = mim1 mim2 mim3 fed1
 
-rock:
-	@if [ "$(FILE)" ]; then elvis rock $(FILE);\
-	elif [ "$(BRANCH)" ]; then tools/rock_changed.sh $(BRANCH); \
-	else tools/rock_changed.sh; fi
-
-run: deps compile quickrun
+# Top-level targets aka user interface
 
-quickrun: etc/ejabberd.cfg etc/app.config certs_priv
-	erl -sname mongooseim@localhost -setcookie ejabberd \
-		-pa ebin deps/*/ebin apps/*/ebin -config etc/app.config \
-		-s mongooseim
+all: rel
 
-etc/ejabberd.cfg:
-	@mkdir -p $(@D)
-	tools/generate_cfg.es etc/ejabberd.cfg rel/files/ejabberd.cfg
+dev: $(DEVNODES)
 
-etc/app.config:
-	@mkdir -p $(@D)
-	tools/generate_cfg.es etc/app.config rel/files/app.config
+clean:
+	-rm -rf _build
```

Comment:
Previous way of doing 'clean' was much better - this one removes all deps sources and they have to be downloaded again.